K8s 1.30 provider slim s390x

chandramerla opened the pull request:
**What this PR does / why we need it**:
Enables s390x support in kubernetes provider 1.30.
This is required for running s390x based tests spinning a k8s cluster using k8s provider.

**Special notes for your reviewer**:
Earlier PR (created for 1.28 provider) whose comments are addressed in this PR: #1201 
Changes required in respective prow jobs to publish provider are in PR: [#3566](https://github.com/kubevirt/project-infra/pull/3566) 

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [x] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
NONE
```

dhiller commented:
/cc

chandramerla commented:
I've run publish.sh inside the containers used in kubevirtci prow jobs, which build and publish gocli, centos9, k8s provider images and alpine container image. With the changes in publish.sh script in this PR and prow job related changes (In PR https://github.com/kubevirt/project-infra/pull/3566), I've successfully tested publishing of multiarch manifest-list for all the above images to icr.io (as I don't have write access in quay.io). In published multiarch manifest-lists s390x arch based images are built/present for gocli, centos9 and k8s-1.30 (slim).

chandramerla commented:
/sig ci

brianmcarey commented:
/test check-provision-k8s-1.30 
/test check-gocli

chandramerla commented:
@brianmcarey : Could you rerun tests and review this?

brianmcarey commented:
/test check-provision-k8s-1.30
/test check-gocli

brianmcarey commented on cluster-provision/centos9/scripts/vm.sh:
```diff
@@ -184,15 +192,100 @@ if [ "${NUMA}" -gt 1 ]; then
+    ${QEMU_ARGS}"
+fi
+
+# Remove secondary network devices from qemu_system_cmd and move them to qemu_monitor_cmds, so
+# that those devices are later added after VM is started using qemu monitor to avoid
+# primary network interface to be named other than eth0. This is mainly required for s390x, as 
+# otherwise if primary interface is other than eth0, it is not getting the IP from dhcp server.
+qemu_monitor_cmds=()
+IFS=' ' read -r -a qemu_parts <<< "$qemu_system_cmd"
+for ((i = 0; i < ${#qemu_parts[@]}; i++)); do
+  part="${qemu_parts[$i]}"
+  nxtpart="${qemu_parts[$i + 1]}"
+  # Check for secondary network devices and move them to qemu_monitor_cmds
+  if { [ "$part" == "-netdev" ] && [[ "$nxtpart" == *"secondarynet"* ]]; } || \
+     { [ "$part" == "-device" ] && [[ "$nxtpart" == *"virtio-net-ccw"* ]] && [[ "$nxtpart" == *"secondarynet"* ]]; }; then
+    qemu_system_cmd=$(echo "$qemu_system_cmd" | sed "s/ -$part $nxtpart//")
+    qemu_monitor_cmds+=("${part}_add $nxtpart")
+  fi
+done
```
This feels strange - we first create the `qemu_system_cmd` and then we remove the secondary network devices? Could we just not include them when creating `qemu_system_cmd` ?

brianmcarey commented on cluster-provision/gocli/cmd/utils/images.go:
```diff
@@ -4,5 +4,5 @@ const (
 	NFSGaneshaImage = "docker.io/janeczku/nfs-ganesha@sha256:17fe1813fd20d9fdfa497a26c8a2e39dd49748cd39dbb0559df7627d9bcf4c53"
 	// DockerRegistryImage contains the reference to docker registry docker image
-	DockerRegistryImage = "quay.io/libpod/registry:2.7"
+	DockerRegistryImage = "quay.io/libpod/registry:2.8.2"
```
Thanks for this but why is this needed for s390x ? We should try to minimize our changes here as the PR is already quite big.

brianmcarey commented on cluster-provision/gocli/cmd/utils/utils.go:
```diff
@@ -34,3 +34,12 @@ func appendIfExplicit(ports nat.PortMap, exposedPort int, flagSet *pflag.FlagSet
 	return nil
 }
+
+// GetSSHUserByArchitecture returns the SSH user
```
I don't think this comment is needed here.

brianmcarey commented on cluster-provision/k8s/1.30/extra-pre-pull-images:
```diff
@@ -2,10 +2,9 @@ quay.io/kubevirtci/install-cni:1.15.0
-quay.io/calico/pod2daemon-flexvol:v3.18.0
+quay.io/calico/cni:v3.27.2
+quay.io/calico/kube-controllers:v3.27.2
+quay.io/calico/node:v3.27.2
```
Could the bump of calico be split out to a different PR? is this needed for the s390x provider?

brianmcarey replied:
The calico bump should be handled in a separate PR - as we will need to do this for all providers to keep it consistent.

chandramerla replied:
ok. I'll start a separate PR for calico bump. 
Just to confirm my understanding - we keep (or try to keep)  same version of calico across all provider versions and update it across provider versions when necessary. Isn't it?

brianmcarey replied:
Yes unless there is a specific reason to have an older version of calico on a certain provider but I don't think there is any reason in this case

chandramerla replied:
Done. create separate PR https://github.com/kubevirt/kubevirtci/pull/1281/files

brianmcarey commented on cluster-provision/k8s/1.30/k8s_provision.sh:
```diff
@@ -62,15 +64,25 @@ function pull_container_retry() {
-cat << EOF >/etc/yum.repos.d/devel_kubic_libcontainers_stable_cri-o_${CRIO_VERSION}.repo
+if [ "$arch" == "s390x" ]; then
+  export CRIO_VERSION=1.28
+  #As crio version available in kubevirtci-crio-mirror/isv_kubernetes_addons_cri-o_stable_v1.30 is broken with issue https://github.com/containers/crun/issues/1494, which is fixed, but yet to part of crio release
```
Is this still true? I saw there were recent crio releases?

chandramerla replied:
Not sure if still true, I've tried a month back. So, let me once try with latest available crio [v1.30.5](https://github.com/cri-o/cri-o/releases/tag/v1.30.5) (or probably even I'll with v1.31.0 if 1.30 doesn't work)

chandramerla replied:
reverted

brianmcarey commented on cluster-provision/k8s/1.30/provision.sh:
```diff
@@ -58,8 +60,16 @@ dnf install -y container-selinux
-dnf install -y openvswitch2.16
+#openvswitch for s390x is not available from the centos default repos.
+if [ "$ARCH" == "s390x" ]; then
+  dnf install -y https://kojipkgs.fedoraproject.org//packages/openvswitch/2.16.0/2.fc36/s390x/openvswitch-2.16.0-2.fc36.s390x.rpm
```
Would there be much needed to get this in the default repos?

chandramerla replied:
I'm not sure. But, I will try finding how to do this and whom to contact, etc. If you know whom to reach, please let me know.

chandramerla replied:
I've asked the question in fedora-devel group https://matrix.to/#/!DMptnKvPTdgoULzUTV:fedoraproject.org/$rbqZqn1e4e5pIJ_UAnjr5xnJJor4c5M3RDFF-fMU5sk?via=fedoraproject.org&via=fedora.im&via=matrix.org

chandramerla replied:
@brianmcarey : I checked with community, basically there are no builds created by [CentOS nfv SIG](https://sigs.centos.org/nfv/) community for s390x. So, better I build from [sources](https://git.centos.org/rpms/openvswitch/branches?branchname=c8s-sig-nfv-openvswitch-2.16) (or better from [srpm](https://mirror.stream.centos.org/SIGs/9-stream/nfv/source/openvswitch-2/Packages/o/)), which has spec file to build an rpm.

chandramerla replied:
The current place from where I'm downloading is also from authentic user releng (it is fedora's release engineering), so I'm thinking may be as a follow-up PR I can fix building it from sources, if you are ok. Otherwise also let me know I will try building rpm from sources. I still need to figure out what exact commands/flags need to pass to rpm-build/moc, to be consistent with that of x86.

brianmcarey replied:
Yes I followed the conversation over on the fedora matrix - this should be ok for now.

brianmcarey commented on cluster-up/check.sh:
```diff
@@ -49,3 +54,7 @@ if is_enabled "$KVM_NESTED"; then
 fi
+
+if is_enabled "$KVM_HPAGE" && [ "$(uname -m)" = "s390x" ]; then
+	echo "[ERR ] $KVM_HPAGE KVM hugepage enabled. It need to be disabled while nested virtualization is enabled for s390x"
```
```suggestion
	echo "[ERR ] $KVM_HPAGE KVM hugepage enabled. It needs to be disabled while nested virtualization is enabled for s390x"
```

chandramerla replied:
Thanks, corrected!

brianmcarey commented on publish.sh:
```diff
@@ -1,8 +1,15 @@
 #!/bin/bash
 
 set -e
+# Uncomment to run/debug in environment where docker is available instead of podman
+#podman() {
+#    docker "$@"
+#}
```
This is probably not needed.

chandramerla replied:
Removed

brianmcarey reviewed (commented):
thanks for this @chandramerla  - I have a couple of comments and questions.

chandramerla reviewed (commented):
@brianmcarey I've replied to your comments/questions.

brianmcarey commented on cluster-provision/k8s/1.30/manifests/local-volume.yaml:
```diff
@@ -113,7 +113,7 @@ spec:
                 fieldPath: metadata.namespace
           - name: JOB_CONTAINER_IMAGE
-            value: "quay.io/external_storage/local-volume-provisioner:v2.3.2"
+            value: "quay.io/external_storage/local-volume-provisioner:v2.5.0"
```
Can we apply this bump to all three providers in a separate PR?

chandramerla replied:
@brianmcarey a new PR created #1285

brianmcarey commented:
@chandramerla can you create an optional presubmit `
check-provision-k8s-1.30-s390x` so that we can test this?

chandramerla commented:
> @chandramerla can you create an optional presubmit ` check-provision-k8s-1.30-s390x` so that we can test this?

@brianmcarey Created the PR https://github.com/kubevirt/project-infra/pull/3681/files

brianmcarey commented:
/test ?

brianmcarey commented:
/test check-provision-k8s-1.30-s390x
/test check-provision-k8s-1.31
/test check-provision-k8s-1.30
/test check-provision-centos-base

brianmcarey commented:
/test check-provision-k8s-1.30-s390x

brianmcarey commented:
/test check-provision-k8s-1.30-s390x

brianmcarey commented:
/test check-provision-k8s-1.30-s390x

brianmcarey commented:
/test check-provision-k8s-1.30-s390x
/test check-provision-k8s-1.31
/test check-provision-k8s-1.30
/test check-provision-centos-base

brianmcarey commented:
/test check-provision-k8s-1.30-s390x
/test check-provision-k8s-1.31
/test check-provision-k8s-1.30
/test check-provision-centos-base

brianmcarey commented:
/test check-provision-k8s-1.30-s390x
/test check-provision-k8s-1.31
/test check-provision-k8s-1.30
/test check-provision-centos-base

brianmcarey commented:
/test check-provision-k8s-1.30-s390x
/test check-provision-k8s-1.31
/test check-provision-k8s-1.30
/test check-provision-centos-base

brianmcarey commented:
/test check-provision-k8s-1.30-s390x
/test check-provision-k8s-1.31
/test check-provision-k8s-1.30
/test check-provision-centos-base

brianmcarey commented:
/test check-provision-k8s-1.30-s390x

brianmcarey commented:
/test check-provision-k8s-1.30-s390x

vamsikrishna-siddu commented:
/test check-provision-k8s-1.30-s390x

vamsikrishna-siddu commented:
/test check-provision-k8s-1.30-s390x

brianmcarey commented:
/test check-provision-k8s-1.30-s390x

vamsikrishna-siddu commented:
/test check-provision-k8s-1.30-s390x

vamsikrishna-siddu commented:
/test check-provision-k8s-1.30-s390x

brianmcarey commented:
/test check-provision-k8s-1.30-s390x

brianmcarey commented:
/test check-provision-k8s-1.30-s390x

brianmcarey commented:
/test check-provision-k8s-1.30-s390x

vamsikrishna-siddu commented:
/test check-provision-k8s-1.30-s390x check-provision-k8s-1.30

brianmcarey commented:
/test all

brianmcarey commented:
/test all

vamsikrishna-siddu commented:
/test check-provision-k8s-1.30

chandramerla commented:
> @chandramerla I've added the following base image with your changes: `quay.io/kubevirtci/centos9:2410230922-f2ef2f72`

Thanks, @brianmcarey . I've Updated the image in the cluster-provision/k8s/base-image. 
Could you please run all jobs again incl. check-provision-k8s-1.30-s390x ?

brianmcarey commented:
/test all

/test check-provision-k8s-1.30-s390x

brianmcarey reviewed (commented):
Looks good @chandramerla  - just a  couple of questions on the publish.

chandramerla commented on publish.sh:
```diff
@@ -44,55 +47,81 @@ function run_provision_manager() {
-    skopeo copy "docker://${TARGET_REPO}/k8s-$i:${PREV_KUBEVIRTCI_TAG}" "docker://${TARGET_REPO}/k8s-$i:${KUBEVIRTCI_TAG}"
-    skopeo copy "docker://${TARGET_REPO}/k8s-$i:${PREV_KUBEVIRTCI_TAG}-slim" "docker://${TARGET_REPO}/k8s-$i:${KUBEVIRTCI_TAG}-slim"
+    skopeo copy "docker://${TARGET_REPO}/k8s-$i:${PREV_KUBEVIRTCI_TAG}-${ARCH}" "docker://${TARGET_REPO}/k8s-$i:${KUBEVIRTCI_TAG}-${ARCH}"
+    skopeo copy "docker://${TARGET_REPO}/k8s-$i:${PREV_KUBEVIRTCI_TAG}-slim-${ARCH}" "docker://${TARGET_REPO}/k8s-$i:${KUBEVIRTCI_TAG}-slim-${ARCH}"
```
Thanks for pointing this.

I think this can be fixed as below - When this script run on x86 first, we are re-tagging only multi-arch manifest. Next, when run on s390x, s390x-arch image will be retagged.
```
  for i in ${IMAGES_TO_RETAG[@]}; do
    echo "INFO: retagging $i (previous tag $PREV_KUBEVIRTCI_TAG)"
    if [ $ARCH == "amd64" ]; then 
      skopeo copy "docker://${TARGET_REPO}/k8s-$i:${PREV_KUBEVIRTCI_TAG}" "docker://${TARGET_REPO}/k8s-$i:${KUBEVIRTCI_TAG}"
      skopeo copy "docker://${TARGET_REPO}/k8s-$i:${PREV_KUBEVIRTCI_TAG}-slim" "docker://${TARGET_REPO}/k8s-$i:${KUBEVIRTCI_TAG}-slim"
    elif [ $ARCH == "s390x" ] && [ $i == "1.30" ]; then
      skopeo copy "docker://${TARGET_REPO}/k8s-$i:${PREV_KUBEVIRTCI_TAG}-${ARCH}" "docker://${TARGET_REPO}/k8s-$i:${KUBEVIRTCI_TAG}-${ARCH}"
      skopeo copy "docker://${TARGET_REPO}/k8s-$i:${PREV_KUBEVIRTCI_TAG}-slim-${ARCH}" "docker://${TARGET_REPO}/k8s-$i:${KUBEVIRTCI_TAG}-slim-${ARCH}"
    fi
  done
```

chandramerla reviewed (commented):
Thanks Brian for the review. I've replied to your comments.

brianmcarey commented:
/test all

/test check-provision-k8s-1.30-s390x

vamsikrishna-siddu commented:
/test check-provision-k8s-1.29

vamsikrishna-siddu commented:
/test all

vamsikrishna-siddu commented:
/test check-provision-k8s-1.30-s390x

vamsikrishna-siddu commented:
/retest

chandramerla commented:
/test all
/test check-provision-k8s-1.30-s390x

brianmcarey commented:
/test all
/test check-provision-k8s-1.30-s390x

chandramerla commented:
/test all
/test check-provision-k8s-1.30-s390x

vamsikrishna-siddu commented:
/test all

vamsikrishna-siddu commented:
/test check-provision-k8s-1.30-s390x

chandramerla commented:
/test check-gocli

chandramerla commented:
/test check-provision-k8s-1.30-s390x

chandramerla commented:
/test check-provision-k8s-1.29

brianmcarey reviewed (commented):
/approve

thanks @chandramerla  - can we get a followup PR as soon as possible to apply these changes to the k8s-1.31 provider too so that they can be automatically included when we create the k8s-1.32 provider shortly?

dhiller commented on cluster-provision/gocli/cmd/run.go:
```diff
@@ -574,19 +575,37 @@ func run(cmd *cobra.Command, args []string) (retErr error) {
+
+	// Use virtio-net-ccw device incase of s390x Architecture.
+	if arch == "s390x" {
+		qemuDevice = "virtio-net-ccw"
```
Could we introduce constants that tell the reader what this is about?

i.e.

```suggestion
const (
    QEMU_DEVICE_S390X  = "virtio-net-ccw"
    QEMU_DEVICE_X86_64 = "virtio-net-pci"
)
```

and use these? Or create a func similar to `GetSSHUserForArchitecture`?

chandramerla replied:
Done

dhiller commented on cluster-provision/gocli/cmd/run.go:
```diff
@@ -574,19 +575,37 @@ func run(cmd *cobra.Command, args []string) (retErr error) {
+			// Secondary network devices are added after VM is started (hot-plug) using qemu monitor to avoid 
+			// primary network interface to be named other than eth0. This is mainly required for s390x, as 
+		    // otherwise if primary interface is other than eth0, it can't get the IP from dhcp server.
+			if qemuDevice == "virtio-net-ccw" {
```
see above, then this would be more obvious.

chandramerla replied:
Yes, updated.

dhiller commented on cluster-up/check.sh:
```diff
@@ -26,12 +26,17 @@ fi
 	KVM_NESTED=$( cat /sys/module/kvm_amd/parameters/nested )
 	KVM_ARCH="amd"
+elif [ -f "/sys/module/kvm/parameters/nested" ]; then
+    KVM_NESTED=$( cat /sys/module/kvm/parameters/nested )
```
please use correct indentation (tabs)

chandramerla replied:
Done

dhiller commented on cluster-up/cluster/ephemeral-provider-common.sh:
```diff
@@ -199,6 +199,8 @@ function _add_common_params() {
     if [ "$KUBEVIRT_DEPLOY_CDI" == "true" ]; then
         params=" --deploy-cdi $params"
+    else
+        params=" --deploy-cdi=false $params"
```
This shouldn't be necessary - iiuc the default for cdi deployment is flipped inside kubevirt/kubevirt: https://github.com/kubevirt/kubevirt/blob/0e54154a8a95745aa7cd54ccd03597c37b1269ed/hack/cluster-up.sh#L20

chandramerla replied:
Yes, with changes from this [commit](https://github.com/kubevirt/kubevirtci/commit/22e4513dd963abd45332f2972e3021423b699d6d#diff-12e7b6e3d4ce92e7d50b6cd0e3aebac95e9480cb4dbbe4a220e1c86c3b12f6ebL124), I don't need to handle this here anymore. 
P.S: Earlier when discussing with Ammar, he suggested to change it here than changing the default in the run.go as that was the default in the shell scripts before he migrated.

Thanks for pointing this.

dhiller commented on publish.sh:
```diff
@@ -44,55 +47,86 @@ function run_provision_manager() {
   (cd cluster-provision/centos9 && ./build.sh)
   IMAGE_TO_BUILD="$(find cluster-provision/k8s/* -maxdepth 0 -type d -printf '%f\n' | head -1)"
+  if [ $ARCH == "s390x" ]; then
+    IMAGE_TO_BUILD="1.30"
```
When will this be removed? After this PR has landed?

chandramerla replied:
I've removed this in [PR](https://github.com/kubevirt/kubevirtci/pull/1316) for 1.31 slim provider supporting s390x.

dhiller reviewed (commented):
@chandramerla I have added a couple of questions inline.

FYI @brianmcarey

chandramerla commented on cluster-provision/gocli/cmd/utils/utils.go:
```diff
@@ -34,3 +34,10 @@ func appendIfExplicit(ports nat.PortMap, exposedPort int, flagSet *pflag.FlagSet
 	return nil
 }
+
+func GetSSHUserByArchitecture(arch string) string {
```
Actually when i've written this function, pkg/libssh/ssh.go didn't exist. But later I didn't realise that we can move it here. Thanks for pointing this, now I've moved it accordingly.

chandramerla commented on cluster-provision/gocli/cmd/provision.go:
```diff
@@ -228,13 +230,14 @@ func provisionCluster(cmd *cobra.Command, args []string) (retErr error) {
+	logrus.Info("Wait for ssh.sh script to exist")
 	err = _cmd(cli, nodeContainer(prefix, nodeName), "while [ ! -f /ssh_ready ] ; do sleep 1; done", "checking for ssh.sh script")
 	if err != nil {
+		logrus.Info("Error: Wait for ssh.sh script to exist")
```
Yes, removed the log

chandramerla commented:
/test check-provision-k8s-1.30

chandramerla commented:
/test check-provision-k8s-1.30-s390x

chandramerla commented:
> @chandramerla I have added a couple of questions inline.
> 
> FYI @brianmcarey

Thanks @dhiller for your comments/questions. Addressed all of them.

dhiller commented on cluster-provision/gocli/cmd/run.go:
```diff
@@ -956,14 +973,17 @@ func provisionNode(sshClient libssh.Client, n *nodesconfig.NodeLinuxConfig) erro
-		bvfio := bindvfio.NewBindVfioOpt(sshClient, s)
-		opts = append(opts, bvfio)
+	// sound cards are not supported on s390x.
+    if runtime.GOARCH != "s390x" {
```
please use tabs for indentation

chandramerla replied:
Thanks. Actually I've couple of more places, where I fixed this indentation issue after you pointed this. I will take of such issues in future.

dhiller replied:
Not sure why spaces slippped in here - I remember that gofmt emits tabs as default

chandramerla replied:
right. Not sure, but today I've (re-)installed all gotools and now I see it is working.

dhiller reviewed (changes requested):
@chandramerla nearly there - once the indentation has slipped, everything else looks fine!

dhiller commented:
@chandramerla would you mind taking a look at the failing check-provision lanes please?

chandramerla commented:
> @chandramerla would you mind taking a look at the failing check-provision lanes please?

@dhiller Thanks for the labels.
I'm seeing all the check-provision jobs (even [outside](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirtci/1317/check-provision-k8s-1.29/1853650013293383680) this PR) are failing with same kubevirt conformance tests related to sig-network.

The common error across all those tests is

> /usr/bin/virt-chroot create-tap --tap-name tap0 --uid 107 --gid 107 --queue-number 0 --mtu 1480, err: exit status 1, output: Error: failed to create tap device named tap0. Reason: [0]: operation not permitted

This is happening since last evening/night, before that I've not seen those sig-network failures.
I don't have much clue here. So I was trying to reach @brianmcarey  on the same.

dhiller commented:
> > @chandramerla would you mind taking a look at the failing check-provision lanes please?
> 
> @dhiller Thanks for the labels. I'm seeing all the check-provision jobs (even [outside](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirtci/1317/check-provision-k8s-1.29/1853650013293383680) this PR) are failing with same kubevirt conformance tests related to sig-network.
> 
> The common error across all those tests is
> 
> > /usr/bin/virt-chroot create-tap --tap-name tap0 --uid 107 --gid 107 --queue-number 0 --mtu 1480, err: exit status 1, output: Error: failed to create tap device named tap0. Reason: [0]: operation not permitted
> 
> This is happening since last evening/night, before that I've not seen those sig-network failures. I don't have much clue here. So I was trying to reach @brianmcarey on the same.

@EdDev @ormergi do you have an idea on why this is happening - why is the tap getting a not permitted error?

dhiller commented:
What we noticed is that the last change to kubevirtci happened on 29th Oct, where the failure started happening yesterday.

dhiller commented:
So it might seem some change on kubevirt itself is responsible here.

EdDev commented:
> > The common error across all those tests is
> > > /usr/bin/virt-chroot create-tap --tap-name tap0 --uid 107 --gid 107 --queue-number 0 --mtu 1480, err: exit status 1, output: Error: failed to create tap device named tap0. Reason: [0]: operation not permitted
> > 
> > 
> > This is happening since last evening/night, before that I've not seen those sig-network failures. I don't have much clue here. So I was trying to reach @brianmcarey on the same.
> 
> @EdDev @ormergi do you have an idea on why this is happening - why is the tap getting a not permitted error?

I do not know what has changed to trigger this, but I do not think it is related to the k/k project.

The permission problem can be caused by SELinux or chroups. These are the reasons I saw in the past.
I suggest you run this one on the kubevirt/kubevirt project.

ormergi commented:
I noticed the new provider use generic cloud image instead of vagrant box for the cluster node OS
Maybe its worth trying with vagrant box to eliminate this is the issue?

Also, I saw the kernel args are slight diffrent, for example the s390x provider kernel args doesnt specify `nopti`, maybe it case some slowness when tap device is created when selinux is involved.
https://github.com/kubevirt/kubevirtci/blob/5481bd92fa3804fad0f14821cda8aff5ae209f24/cluster-provision/centos9/scripts/kernel.s390x.args
https://github.com/kubevirt/kubevirtci/blob/5481bd92fa3804fad0f14821cda8aff5ae209f24/cluster-provision/centos9/scripts/kernel.args

brianmcarey commented:
> I noticed the new provider use generic cloud image instead of vagrant box for the cluster node OS Maybe its worth trying with vagrant box to eliminate this is the issue?
> 
> Also, I saw the kernel args are slight diffrent, for example the s390x provider kernel args doesnt specify `nopti`, maybe it case some slowness when tap device is created when selinux is involved. https://github.com/kubevirt/kubevirtci/blob/5481bd92fa3804fad0f14821cda8aff5ae209f24/cluster-provision/centos9/scripts/kernel.s390x.args https://github.com/kubevirt/kubevirtci/blob/5481bd92fa3804fad0f14821cda8aff5ae209f24/cluster-provision/centos9/scripts/kernel.args

Hi @ormergi  - thanks for taking a look at this - we are seeing this on all kubevirtci PRs since the 4th. For example a simple PR - https://github.com/kubevirt/kubevirtci/pull/1317

ormergi commented:
> Hi @ormergi - thanks for taking a look at this - we are seeing this on all kubevirtci PRs since the 4th. For example a simple PR - #1317

Thanks for the heads up
I see last change was centoOS bump https://github.com/kubevirt/kubevirtci/pull/1314, its not the first time we see issues with such changes, lets try to revert this change and see if things stabilize, WDTY?

brianmcarey commented:
/hold 

To avoid needless retesting.

brianmcarey commented:
/hold cancel 

/retest

chandramerla commented:
> Hi, since then the changes were populated to k8s versions > 1.30 ? This PR added changes to 1.30 only Thanks

Hi @oshoval 

Yes, apart from these changes for 1.30, some changes were done in 1.31 https://github.com/kubevirt/kubevirtci/pull/1316/files.
With that, all following versions like 1.32 (which made a copy from 1.31) have the changes.
Hope this answers your question.

oshoval commented:
Hi @chandramerla 
Yep thank you
